### ISSUE TITLE
Make it easy for users to share application logs and diagnostic info when reporting bugs

### ISSUE THREAD

**Issue by gnarea** [relaycorp/relaynet-gateway-android#262]
Android users already get an option to report crashes to the developer, but not all bugs result in crashes and therefore we currently don't have a way to diagnose those.

For example, I was confused to find that the app said I wasn't connected to Relaynet (screenshot below), when in fact the phone was connected to the Internet and the public gateway was accessible. (I had to restart the app to fix this)

![Screenshot_20210114-153257](https://user-images.githubusercontent.com/369863/104615840-865ddb80-5681-11eb-908e-d5ec4884a5aa.png)

However we do this, we need a way to allow users to send logs and other info (e.g., Android version, device model, private gateway address). We could, for instance, create a text file that the user will be able to share via email with us (though if we could somehow leverage Android crash reports, that'd be even better).

**Comment by sdsantos** [comment]
We can't leverage Android crash reports for this, it's a closed system.

We either use a third-party service for this (which we likely don't want), or we need to write our own log storage system, and add a send email button that attaches the app logs.

**Comment by gnarea** [comment]
Indeed, I wouldn't send that data to a third-party service. I guess extracting the app logs and sending it via email (along with some system info) is the way to go.
